middleware/http/opa/middleware.go

```diff
@@ -15,7 +16,7 @@ import (
 
 type middlewareMetadata struct {
 	Rego            string `json:"rego"`
-	DefaultStatus   int    `json:"defaultStatus,omitempty"`
+	DefaultStatus   int    `json:"-"`
```

Comment:
Instead of splitting the marshalling between annotation and the special handling in `getNativeMetadata` consider keeping this annotation and replacing the special handling in `getNativeMetadata` with a custom `UnmarshalJSON` handler for clarity:

```go

func (mm *middlewareMetadata) UnmarshalJSON(b []byte) error {
	// For brevity, assume we only unmarshal from metadata string map.
	// If middlewareMetadata is itself roundtripped through JSON, need 
	// to handle both string and int mappings to DefaultStatue.
	var v map[string]string
	if err := json.Unmarshal(b, &v); err != nil {
		return err
	}
	mm.Rego = v["rego"]
	mm.IncludedHeaders = v["includedHeaders"]

	if ds, ok := v["defaultStatus"]; ok {
		if intDS, intErr := strconv.Atoi(ds); intErr != nil {
			return intErr
		} else {
			mm.DefaultStatus = intDS
		}
	}
	return nil
}
```

Reply:
Good! Done.

Reply:
I defined an alias type of int for defaultStatus. And define with a custom UnmarshalJSON handler for it.